modules/geoipbackend/geoipbackend.cc

```diff
@@ -309,104 +305,129 @@ string GeoIPBackend::queryGeoIP(const string &ip, bool v6, GeoIPQueryAttribute a
   GeoIPRecord *gir2 = NULL;
   int id;
 
-  if (v6 && s_gi6) {
-    if (attribute == Afi) {
-      return "v6";
-    } else if (d_dbmode == GEOIP_ISP_EDITION_V6 || d_dbmode == GEOIP_ORG_EDITION_V6) {
-      if (attribute == Name) {
-        val = GeoIP_name_by_addr_v6_gl(s_gi6, ip.c_str(), gl);
-      }
-    } else if (d_dbmode == GEOIP_COUNTRY_EDITION_V6 ||
-        d_dbmode == GEOIP_LARGE_COUNTRY_EDITION_V6 ||
-        d_dbmode == GEOIP_COUNTRY_EDITION) {
-      id = GeoIP_id_by_addr_v6_gl(s_gi6, ip.c_str(), gl);
-      if (attribute == Country) {
-        val = GeoIP_code3_by_id(id);
-      } else if (attribute == Continent) {
-        val = GeoIP_continent_by_id(id);
-      }
-    } else if (d_dbmode == GEOIP_REGION_EDITION_REV0 ||
-        d_dbmode == GEOIP_REGION_EDITION_REV1) {
-      gir = GeoIP_region_by_addr_v6_gl(s_gi6, ip.c_str(), gl);
-      if (gir) {
+  for(auto const& gi: s_geoip_files) {
+    if (v6) {
+      if (attribute == Afi) {
+        return "v6";
+      } else if (gi.first == GEOIP_ISP_EDITION_V6 || gi.first == GEOIP_ORG_EDITION_V6) {
+        if (attribute == Name) {
+          val = GeoIP_name_by_addr_v6_gl(gi.second.get(), ip.c_str(), gl);
+        }
+      } else if (gi.first == GEOIP_ASNUM_EDITION_V6) {
+        if (attribute == ASn) {
+          vector<string> asnr;
+          val = GeoIP_name_by_addr_v6_gl(gi.second.get(), ip.c_str(), gl);
+          if (val) {
+            stringtok(asnr, val);
+            if(asnr.size()>0) {
+              val = asnr[0].c_str();
```

Comment:
Since you are doing this you *must* move the vector<string> ansr to the scope of this method, otherwise this is very unsafe thing to do as vector<string> asnr; gets deleted at the end of this if.